### PR TITLE
infosync: integrate PD HTTP client into the placement manager

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #35319, https://github.com/tikv/pd/issues/7300

Problem Summary:

Integrate PD HTTP client into the placement manager.

### What is changed and how it works?

- Replace the etcd client inside `PlacementManager` with PD HTTP client.
- Reuse the type definitions within the PD SDK as much as possible.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/placement/rule.go [hunk 1]
```diff
@@ -302,15 +277,3 @@ func newRulesWithDictConstraints(role PeerRoleType, cnstr string) ([]*Rule, erro
...
-// Note that it is a shallow copy: Constraints is not cloned.
-func (r *Rule) Clone() *Rule {
-	n := &Rule{}
-	*n = *r
-	return n
-}
-
-func (r *Rule) String() string {
-	return fmt.Sprintf("%+v", *r)
-}
```
Comment (JmPotato) [inline pkg/ddl/placement/rule.go:316]: These two methods are implemented here now:

https://github.com/tikv/pd/blob/58e9b2088b069a2c6a1e58be9dac752e46593e44/client/http/types.go#L259-L272

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**